### PR TITLE
feature/mx-1702 resolve types in edit view

### PR DESCRIPTION
**Pull request by cutoffthetop**
# PR Context
- this PR only resolves Links, Texts and Enums, but does not resolve Identifiers yet
- additionally cleans up the user interface and code structure a bit
- needs https://github.com/robert-koch-institut/mex-backend/pull/187 and https://github.com/robert-koch-institut/mex-backend/pull/189

# Changes
- upgrade mex-common and model dependencies to v3
- overhaul and simplify margins and spaces
- move transform_models_to_fields from State to transform module
- use dedicated backend connector methods for edit and search
- use same rendering components for edit and search pages
- pop toasts when backend connector encounters errors
- scroll to top when pagination is triggered

# Fixed
- fix routing issues by moving the refresh handlers section.from on_mount to page.on_load

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +26,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- fix routing issues by moving the refresh handlers section.from on_mount to page.on_load
```
Comment (rababerladuseladim) [inline CHANGELOG.md]:
```suggestion
- fix routing issues by moving the refresh handlers section from `on_mount` to `page.on_load`
```

File: mex/editor/components.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    return rx.cond(
+        value.external,
+        fixed_external_link(value),
+        fixed_internal_link(value),
+    )
+
+
+def fixed_text(value: FixedValue) -> rx.Component:
+    """Render a fixed value as a text span with language attribute."""
+    return rx.text(
```
Comment (rababerladuseladim) [inline mex/editor/components.py:38]: Where is the language attribute?

File: pyproject.toml [hunk 3]
```diff
@@ -10,22 +10,26 @@ requires-python = ">=3.11,<3.12"
...
-    "reflex>=0.6,<1",
-    "starlette>=0.39,<1",
+    "reflex>=0.5,<0.6.3",
+    "requests>=2,<3",
+    "starlette>=0.41,<1",
     "uvicorn>=0.32,<1",
 ]
 optional-dependencies.dev = [
     "ipdb>=0.13,<1",
+    "mex-backend@git+https://github.com/robert-koch-institut/mex-backend.git@0.22.0",
```
Comment (rababerladuseladim) [inline pyproject.toml:25]: Why does the editor depend on the backend?

File: mex/editor/transform.py [hunk 4]
```diff
@@ -1,40 +1,128 @@
...
+    TemporalEntityPrecision.HOUR: "d. MMMM yyyy k a",
+    TemporalEntityPrecision.MINUTE: "d. MMMM yyyy H:MM",
+    TemporalEntityPrecision.SECOND: "d. MMMM yyyy H:MM:ss",
+    TemporalEntityPrecision.MICROSECOND: "d. MMMM yyyy H:MM:ss:SS",
+}
+
+
+def transform_values(values: object) -> list[FixedValue]:
+    """Convert a single object or a list of objects into a list of fixed values."""
+    if values is None:
```
Comment (rababerladuseladim) [inline mex/editor/transform.py:33]: Why doesn't this return the same as `transform_value(None)`?

File: mex/editor/transform.py [hunk 5]
```diff
@@ -1,40 +1,128 @@
...
+        return []
+    titles: list[FixedValue] = []
+    for model in models:
+        config = MODEL_CONFIG_BY_STEM_TYPE[model.stemType]
+        titles.extend(
+            transform_values(getattr(model, config.title)),
+        )
+    if titles:
+        return titles
+    return transform_values(models[0].identifier)
```
Comment (rababerladuseladim) [inline mex/editor/transform.py:106]: Please add a test for the no models and no titles case.

File: tests/test_transform.py [hunk 6]
```diff
@@ -1,54 +1,165 @@
...
-    ids=["None", "dict", "list", "Text", "Link", "string-like", "nested"],
-)
-def test_render_any_value(value: Any, expected: str) -> None:
-    assert render_any_value(value) == expected
+from mex.common.models import AnyExtractedModel
+from mex.editor.models import FixedValue
+from mex.editor.transform import transform_models_to_preview, transform_models_to_title
 
 
 def test_render_model_title(dummy_data: list[AnyExtractedModel]) -> None:
```
Comment (rababerladuseladim) [inline tests/test_transform.py]:
```suggestion
def test_transform_models_to_title(dummy_data: list[AnyExtractedModel]) -> None:
```

File: tests/test_transform.py [hunk 7]
```diff
@@ -1,54 +1,165 @@
...
+                badge="de",
+                href=None,
+                tooltip=None,
+                external=False,
+            )
+        ],
     ]
 
 
 def test_render_model_preview(dummy_data: list[AnyExtractedModel]) -> None:
```
Comment (rababerladuseladim) [inline tests/test_transform.py]:
```suggestion
def test_transform_models_to_preview(dummy_data: list[AnyExtractedModel]) -> None:
```

File: tests/test_transform.py [hunk 8]
```diff
@@ -1,54 +1,165 @@
...
-    dummy_previews = [render_model_preview(d) for d in dummy_data]
+    dummy_previews = [transform_models_to_preview([d]) for d in dummy_data]
```
Comment (rababerladuseladim) [inline tests/test_transform.py:136]: please add tests for the no preview / no models case.

File: tests/conftest.py [hunk 9]
```diff
@@ -76,24 +101,57 @@ def login_user(page: Page, username: str, password: SecretStr) -> Page:
...
-    writer_user_credentials: tuple[str, SecretStr],
+    page: Page, writer_user_credentials: tuple[str, SecretStr], frontend_url: str
 ) -> Page:
-    login_user(page, *writer_user_credentials)
+    login_user(frontend_url, page, *writer_user_credentials)
     expect(page.get_by_test_id("nav-bar")).to_be_visible()
     return page
 
 
+class ResettingGraphConnector(GraphConnector):
```
Comment (rababerladuseladim) [inline tests/conftest.py:120]: Meh, any idea how to not depend on mex-backend here?

**Review by rababerladuseladim — CHANGES_REQUESTED** [review]
Only minor nitpicking.

File: pyproject.toml (hunk 3 above)
Comment (cutoffthetop) [reply pyproject.toml:25]: it's a dev-dependency because i'm re-using the graph connector in the tests

File: mex/editor/components.py (hunk 2 above)
Comment (cutoffthetop) [reply mex/editor/components.py:38]: doc fixed c63dde7

File: tests/conftest.py (hunk 9 above)
Comment (cutoffthetop) [reply tests/conftest.py:120]: we could move the graph connector to mex-common as an extra? a la `pip install mex-common[graph]`

File: mex/editor/transform.py (hunk 4 above)
Comment (cutoffthetop) [reply mex/editor/transform.py:33]: fixed in 7fbb1c5 and tested in ec64850

File: mex/editor/transform.py (hunk 5 above)
Comment (cutoffthetop) [reply mex/editor/transform.py:106]: i added a simple no-models test. no-titles was already covered below, but i added comments in d6beb76

File: tests/test_transform.py (hunk 8 above)
Comment (cutoffthetop) [reply tests/test_transform.py:136]: 550a942

File: tests/conftest.py (hunk 9 above)
Comment (cutoffthetop) [reply tests/conftest.py:120]: if you see that as an option too, i'd like to push it to a new ticket though. i drafted MX-1746, so we can discuss in refinement

**Review by rababerladuseladim — APPROVED** [review]

**Pull request merged**